pkgs/development/tools/analysis/retdec/yaracpp.nix

```diff
@@ -0,0 +1,49 @@
+{ stdenv
+, fetchFromGitHub
+, coreutils
+}:
+
+let
+  yara = fetchFromGitHub {
+    owner = "avast-tl";
+    repo = "yara";
+    rev = "ea101c5856941f39cad2db3012f2660d1d5c8b65";
+    sha256 = "033ssx2hql5k4pv9si043s3mjq2b748ymjzif8pg6rdwh260faky";
+  };
+in stdenv.mkDerivation rec {
+  # only fetches the yaracpp source patched to work with a local yara clone,
+  # does not build anything
+  name = "yaracpp-src-${version}";
+  version = "2018-10-09";
+  rev = "b92bde0e59e3b75bc445227e04b71105771dee8b"; # as specified in retdec/deps/yaracpp/CMakeLists.txt
+
+  src = fetchFromGitHub {
+    inherit rev;
+    owner = "avast-tl";
+    repo = "yaracpp";
+    sha256 = "0fan7q79j7s3bjmhsd2nw6sqyi14xgikn7mr2p4nj87lick5l4a2";
+  };
+
+  postPatch = ''
+      # check if our version of yara is the same version that upstream expects
+      echo "Checking version of yara"
+      expected_rev="$( sed -n -e 's|.*URL https://github.com/.*/archive/\(.*\)\.zip.*|\1|p' "deps/CMakeLists.txt" )"
+      if [ "$expected_rev" != '${yara.rev}' ]; then
+        echo "The yara dependency has the wrong version: ${yara.rev} while $expected_rev is expected."
+        exit 1
+      fi
```

Comment:
It seems strange to compare strings in different quoting styles (single quotes around `yara.rev`)? Kinda surprised it works, but since it seems to... :).

Reply:
I agree, but I don't see a good way to do it either. I don't really like the whole business of putting `yaracpp` in its own file, but I don't see a way around it. I'm naively hoping for upstream help in the future.

Reply:
The quoting is because the `$expected_rev` is a bash variable (expanded within double quotes) while `${yara.rev}` is a nix variable which is expanded before bash gets to see the script. So from bash's point of view it is a string literal, which generally should be in single quotes to avoid accidental expansion of variables. Technically using `shellEscape` would be even better, but in this case its just a hash and shouldn't matter either way.

Reply:
Alright, sounds good then! Just stood out during a spot-check, so a)it works b)you have reasons.. sounds good to me! :)